backend/app/main/views.py

```diff
@@ -272,3 +296,74 @@ def send_mail():
 
     send_email(recipients, subject, body)
     return "message sent!"
+
+
+# log in an existing user
+@main.route("/auth/login", methods=["POST"])
+def login():
+    # required in body: email: String
+
+    data = request.get_json(force=True)
+    email = data.get("email")
+
+    # get mentor
+    mentor = get_mentor_by_email(email)
+
+    # check if valid mentor
+    if mentor[1] == 200:
+        # create user
+        user_info = json.loads(mentor[0].response[0])
+        user = Mentor(
+            name=user_info["name"],
+            email=user_info["email"],
+            id_number=user_info["id"]
+        )
+        # create tokens
+        access_token = create_access_token(identity=user.id_number)
+        refresh_token = create_refresh_token(identity=user.id_number)
+
+        # set cookies
+        resp = jsonify({"user": user.serialize()})
+        set_access_cookies(resp, access_token)
+        set_refresh_cookies(resp, refresh_token)
+
+        return resp, 200
+
+    return "Wrong email or password!", 400
+
+
+@main.route("/auth/token/refresh", methods=["POST"])
+def refresh_token():
+
+    # create the new access token
+    access_token = request.headers["Cookie"].split(";")[0].split("=")[1]
+
+    # get mentor
+    mentor_id = decode_token(access_token)['identity']
+    mentor = get_mentor_by_id(mentor_id)
+
+    # check if valid mentor
+    if mentor[1] == 200:
+        # create user
+        user_info = json.loads(mentor[0].response[0])
+        user = Mentor(
+            name=user_info["name"],
+            email=user_info["email"],
+            id_number=user_info["id"]
+        )
+
+        # set the JWT access cookie in the response
+        resp = jsonify({"user": user.serialize()})
+        set_access_cookies(resp, access_token)
+
+        return resp, 200
+
+    return "Invalid id", 400
```

Comment:
This overall seems a bit janky to me, take a look at https://github.com/kraglalbert/budgie-web/blob/master/backend/app/auth/views.py#L50 and the `refresh()`method in https://flask-jwt-extended.readthedocs.io/en/stable/tokens_in_cookies/ for the library's way of doing it

Reply:
**TODO**: add JWT decorators